compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -344,8 +344,13 @@ pub fn target_features(sess: &Session, allow_unstable: bool) -> Vec<Symbol> {
         })
     {
         if enabled {
+            // Also add all transitively implied features.
             features.extend(sess.target.implied_target_features(std::iter::once(feature)));
         } else {
+            // `!feature` implies `!other_feature` if `other_feature` (transitively) implies `feature`,
```

Comment:
LGTM

IMO, the new comment reads like code still. Looking at the existing comment below; 

> We don't care about the order in `features` since the only thing we use it for is the
    `features.contains` below.

This is the style that I would prefer, as opposed to;

> `X` implies `Y` if `Y` implies `X`

Reply:
I wanted to make the connection to the standard contraposition principle explicit. Not sure how to do that in any other style. ;)